test/tests/mfem/submeshes/tests

```diff
@@ -0,0 +1,37 @@
+[Tests]
+  design = 'syntax/Problem/index.md'
+  issues = '#30478'
+  [./MFEMDomainSubMesh]
+    type = XMLDiff
+    input = domain_submesh.i
+    xmldiff = 'OutputData/DomainPotential/Run0/Run0.pvd
+                OutputData/DomainPotential/Run0/Cycle000001/proc000000.vtu'
+    ignored_items = "root['VTKFile']['@version']"
+    requirement = 'MOOSE shall have the ability to solve a diffusion problem defined on an MFEM domain restricted SubMesh.'
```

Comment:
```suggestion
    requirement = 'MOOSE shall have the ability to solve a diffusion problem defined on an MFEM domain restricted submesh.'
```
We try to keep implementation details out of requirements (other than MFEM because we do have to eliminate hypothetical requirement duplication with a libmesh-based test)